### PR TITLE
MM-53107 Add setting to limit the max number of markdown nodes 

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
We're adding a setting to limit this on mobile to help improve performance, but I figured it would be a good idea to add a setting for this just in case some people are used to using massive, mobile-lagging posts.

Note that I fixed a bug where numeric settings in the System Console were previously unable to be set to 0. I wanted the default value on the server to let the mobile app define its own default since that would make potentially changing the default easier in the future, and there was a minor bug preventing me from using 0 because it was using `||` instead of `??`. I don't think this should cause any problems for existing settings since almost all of the numeric ones have server-side validation already, but I added it to a couple more settings just to be safe.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53107

#### Related Pull Requests
https://github.com/mattermost/commonmark-react-renderer/pull/5
https://github.com/mattermost/mattermost-mobile/pull/7528
Docs PR incoming

#### Release Note
```release-note
Added a setting to limit the maximum complexity of markdown text on mobile
```

### CODE REVIEW

**Comment by marianunez** [comment]
@hmhealey Jesus is OOO this next two weeks so do you want to swap him out given our timeline for the mobile release?

File: webapp/channels/src/components/admin_console/schema_admin_settings.jsx [hunk 1]
```diff
@@ -447,7 +447,7 @@ export default class SchemaAdminSettings extends React.PureComponent {
             inputType = 'textarea';
         }
 
-        let value = this.state[setting.key] || '';
+        let value = this.state[setting.key] ?? '';
```
Comment (M-ZubairAhmed) [inline webapp/channels/src/components/admin_console/schema_admin_settings.jsx:450]: If we are doing optional chaining should we use “this.state?.[setting.key]?? ‘’”

**Review by M-ZubairAhmed — APPROVED** [review]
This looks good :shipit: Thank you

**Review by larkox — APPROVED** [review]

File: webapp/channels/src/components/admin_console/schema_admin_settings.jsx (hunk 1 above)
Comment (hmhealey) [reply webapp/channels/src/components/admin_console/schema_admin_settings.jsx:450]: `this.state` is always defined, so we shouldn't need to. I should've also mentioned that this was the value to have these set to `0`

**Pull request merged**